deeplake/core/chunk/chunk_compressed_chunk.py

```diff
@@ -35,16 +36,194 @@ def __init__(self, *args, **kwargs):
         self._changed = False
         self._compression_ratio = 0.5
 
-    def extend_if_has_space(self, incoming_samples: List[InputSample], update_tensor_meta: bool = True) -> float:  # type: ignore
+    def extend_if_has_space(self, incoming_samples: List[InputSample], update_tensor_meta: bool = True, lengths: Optional[List[int]] = None) -> float:  # type: ignore
         self.prepare_for_write()
+        if lengths is not None:  # this is triggered only for htype == "text"
+            return self.extend_if_has_space_byte_compression_text(
+                incoming_samples, update_tensor_meta, lengths
+            )
         if self.is_byte_compression:
+            if isinstance(incoming_samples, np.ndarray):
+                return self.extend_if_has_space_byte_compression_numpy(
+                    incoming_samples, update_tensor_meta
+                )
             return self.extend_if_has_space_byte_compression(
                 incoming_samples, update_tensor_meta=update_tensor_meta
             )
         return self.extend_if_has_space_image_compression(
             incoming_samples, update_tensor_meta=update_tensor_meta
         )
 
+    def extend_if_has_space_byte_compression_text(
+        self,
+        incoming_samples: np.ndarray,
+        update_tensor_meta: bool = True,
+        lengths: Optional[List[int]] = None,
+    ):
+        sample_nbytes = np.mean(lengths)
+        min_chunk_size = self.min_chunk_size
+        decompressed_bytes = self.decompressed_bytes
+        while True:
+            if sample_nbytes:
+                num_samples = int(
+                    max(
+                        0,
+                        min(
+                            len(incoming_samples),
+                            (
+                                (min_chunk_size / self._compression_ratio)
+                                - len(decompressed_bytes)
+                            )
+                            // sample_nbytes,
+                        ),
+                    )
+                )
```

Comment:
Figure out how many incoming samples we accept. 

Equation for perfect chunk (assuming constant compression ratio )is:

(len(decompressed_bytes) + num_samples * sample_nbytes) * compression_ratio = min_chunk_size

Rearrange to get num_samples alone on the LHS.